test/integration/parser_config.cc

```diff
@@ -0,0 +1,240 @@
+/*
```

Comment:
Are there any tests which explicitly show shadowing / prioritization?
e.g. both a search path and a callback could resolve the same file (possibly to different places), but only one of them wins out?

Reply:
Added test in https://github.com/osrf/sdformat/pull/439/commits/8977cd1111165a0b2f13c902e96f960f96d1cae9